pallets/erc20-xcm-bridge/src/xcm_exec_filter.rs

```diff
@@ -0,0 +1,114 @@
+// Copyright 2019-2022 PureStake Inc.
+// This file is part of Moonbeam.
+
+// Moonbeam is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Moonbeam is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Moonbeam.  If not, see <http://www.gnu.org/licenses/>.
+
+use frame_support::traits::Contains;
+use xcm::latest::prelude::*;
+
+type RuntimeCallOf<T> = <T as frame_system::Config>::RuntimeCall;
+
+/// Morph a given "xcm execution" filter implementation into one which filter out
+/// some XCM instructions if they could manipulate erc20 assets.
+/// If your runtime allows arbitrary xcm messages to be executed locally you should use this
+// wrapper.
+pub struct XcmExecuteFilterWrapper<Runtime, FallbackFilter>(
```

Comment:
In other words the message will fail. This should be no security risk, if something we are telling users not to try this because their message will fail. And I am not a big fan of the latter...